### PR TITLE
RFC: Add static code analyzer integration

### PR DESCRIPTION
**Pull request by romkell**
PR dropped and closed in favor of #51212.

### CODE REVIEW

File: scripts/sca_integration/sca_export.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (romkell) [inline scripts/sca_integration/sca_export.json:1]: The file is a generated file and will be removed from the commit sooner or later.
It is here to illustrate the output format.

Question:
Why are CMAKE_CXX_COMPILER and all the other variables empty except for the CMAKE_C_COMPILER?
Did I include the sca_integration.cmake too early?
Any hint for me?

**Comment by cfriedt** [comment]
@romkell - would be happy to backport, but this would need to go into `main` prior to `v2.7-branch`

**Comment by romkell** [comment]
> @romkell - would be happy to backport, but this would need to go into `main` prior to `v2.7-branch`

@cfriedt With our company internal framework we are currently stuck on a 2.7 tag. Therefore I implemented against that, but tried to keep it as independent as possible to be hooked in at the right location (3.2 cmake files have been rather reorganized compared to 2.7). We will need it on 2.7-auditable-branch. My initial thought was that we would take the latest 2.7 tag as new 2.7-auditable-branch to get all the fixes. Does not work since 2.7-branch and 2.7-auditable-branch are very much out of sync due to misra fixes in auditable.
Hence I will have to take it to 2.7-auditable-branch and also main (on which I would currently have a hard time to test).

Would it be ok to create a PR to main with a not tested, half finished implementation to get the attention and feedback?

The main goal at this time is to get feedback on the mechanism and also to get some hints why basically all CMAKE_<xyz>_COMPILER/LINKER variables are empty (except for CMAKE_C_COMPILER).

**Comment by romkell** [comment]
> If you want to pull this into v2.7-auditable-branch, that is up to you and the Safety Committee; but, it will likely never go into the v2.7-branch or any other previous release branches.

@stephanosio Please consider the v2.7-branch a mistake. Can we discuss the mechansim / solution proposal in this PR or do I need to create one aiming to main (I assume not getting a lot of feedback when aiming for 2.7-auditable-branch)?

**Comment by stephanosio** [comment]
> Can we discuss the mechansim / solution proposal in this PR or do I need to create one aiming to main (I assume not getting a lot of feedback when aiming for 2.7-auditable-branch)?

I would say creating one for main is going to be for the best.

**Comment by romkell** [comment]
PR dropped and closed in favor of #51212.

**Pull request closed without merging**